certbot/src/certbot/achallenges.py

```diff
@@ -43,10 +44,31 @@ class AnnotatedChallenge(jose.ImmutableMap):
     def __getattr__(self, name: str) -> Any:
         return getattr(self.challb, name)
 
+    def __getattribute__(self, name: str) -> Any:
+        if name == 'domain':
+            warnings.warn("the domain attribute is deprecated and will be removed in "
+                        "an upcoming release", DeprecationWarning)
+        return object.__getattribute__(self, name)
```

Comment:
nit:
```suggestion
        return super().__getattribute__(name)
```
it doesn't matter in this case, but i think we should default to calling the parent method unless we have an good reason not to